Library/Formula/upx.rb

```diff
@@ -1,24 +1,24 @@
-require 'formula'
-
 class Upx < Formula
-  homepage 'http://upx.sourceforge.net'
-  url 'http://upx.sourceforge.net/download/upx-3.91-src.tar.bz2'
-  sha1 'da962c0affb27eae11aa9b6fbd751c3699717b36'
-  head 'https://www.pysol.org:4443/hg/upx.hg', :using => :hg
+  homepage "http://upx.sourceforge.net"
+  url "http://upx.sourceforge.net/download/upx-3.91-src.tar.bz2"
+  sha1 "da962c0affb27eae11aa9b6fbd751c3699717b36"
+  head "https://www.pysol.org:4443/hg/upx.hg", :using => :hg
+  revision 1
 
-  depends_on 'ucl'
+  depends_on "ucl"
 
-  resource 'lzma' do
-    url 'https://downloads.sourceforge.net/project/sevenzip/LZMA%20SDK/4.65/lzma465.tar.bz2'
-    sha1 '5ec1c4606fec88c770a9712073e83916f8aed173'
+  resource "lzma" do
+    url "https://downloads.sourceforge.net/project/sevenzip/LZMA%20SDK/lzma938.7z"
+    sha1 "fe7e45f87b0635b453af920dfa15a8ad024b7e96"
   end
 
   def install
-    (buildpath/'lzmasdk').install resource('lzma')
-    ENV['UPX_LZMADIR'] = buildpath/'lzmasdk'
-    ENV['UPX_LZMA_VERSION'] = '0x465'
+    inreplace "src/compress_lzma.cpp", "C/Types.h", "C/7zTypes.h"
+    (buildpath/"lzmasdk").install resource("lzma")
+    ENV["UPX_LZMADIR"] = buildpath/"lzmasdk"
+    ENV["UPX_LZMA_VERSION"] = "0x938"
     system "make all"
-    bin.install  'src/upx.out' => 'upx'
-    man1.install 'doc/upx.1'
+    bin.install  "src/upx.out" => "upx"
+    man1.install "doc/upx.1"
   end
 end
```

Comment:
Can a test be added to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!